otter/util/retry.py

```diff
@@ -340,12 +340,14 @@ class Retry(object):
     :param should_retry: The function to call to determine whether retry
     should occur (usually an instance of :obj:`ShouldDelayAndRetry`).
     """
-    def perform_effect(self, dispatcher):
-        """
-        Invoke :func:`effect.retry.retry` with the effect and the
-        should_retry function.
-        """
-        return effect_retry(self.effect, self.should_retry)
+
+
+def perform_retry(dispatcher, intent):
```

Comment:
> Performers can be implemented by returning Effects instead of actually doing stuff imperatively.

Hmm. I am not sure I completely agree with this or understand this. In any case, this doesn't block the PR :) We can continue this conversation someplace else.